persistence/src/main/scala/com/mesosphere/usi/repository/PodRecordRepository.scala

```diff
@@ -0,0 +1,40 @@
+package com.mesosphere.usi.repository
+
+import com.mesosphere.usi.core.models.{PodId, PodRecord}
+
+import scala.concurrent.Future
+
+/**
+  * Repository for [[PodRecord]].
+  */
+trait PodRecordRepository {
+
+  /**
+    * Stores the provided [[PodRecord]] in the repository if it doesn't exist.
+    * @param record
+    * @return
```

Comment:
What's returned when? This should not be a blocker now but we'll have to document it.